src/transformers/generation/utils.py

```diff
@@ -3755,16 +3755,36 @@ def _gather_beams(tensor: torch.Tensor, beam_indices: torch.Tensor) -> torch.Ten
         return gathered_tensor
 
     @staticmethod
-    def _beam_search_has_unfinished_sequences(
+    def _get_improvement_possibility(
```

Comment:
Request: instead of "improvement"-related variable names and functions, let's use "done"/"complete"-related names.

Technically, the beams can still be improved, if by "improved" we mean "we can update the beams with higher scores", so "improvement"-related names might confuse the reader.

Examples:
`_get_improvement_possibility` -> `_check_if_done`
`is_improvement_possible` -> `is_done`
...